scripts/build/builders/imx.py

```diff
@@ -74,30 +75,58 @@ def __init__(self,
         self.release = release
         self.app = app
 
-    def GnBuildEnv(self):
-        return {
-            'PKG_CONFIG_PATH': self.SysRootPath('IMX_SDK_ROOT') + '/sysroots/cortexa53-crypto-poky-linux/lib/aarch64-linux-gnu/pkgconfig',
-        }
-
     def GnBuildArgs(self):
+        entries = os.listdir(self.SysRootPath('IMX_SDK_ROOT'))
+        for entry in entries:
+            if re.match(r'^environment-setup-', entry):
+                env_setup_script = entry
+                break
+
+        with open(os.path.join(self.SysRootPath('IMX_SDK_ROOT'), env_setup_script), 'r') as env_setup_script_fd:
+            lines = env_setup_script_fd.readlines()
+            for line in lines:
+                line = line.strip('\n')
+                if re.match(r'export SDKTARGETSYSROOT=', line):
+                    sdk_target_sysroot = line[len(r'export SDKTARGETSYSROOT='):]
+                if re.match(r'export CC=', line):
+                    # remove the quotation marks, replace a shell env
+                    cc = line[len(r'export CC=')+1: -1]
+                    cc = cc.replace('$SDKTARGETSYSROOT', sdk_target_sysroot)
+                if re.match(r'export CXX=', line):
+                    # remove the quotation marks, replace a shell env
+                    cxx = line[len(r'export CXX=')+1: -1]
+                    cxx = cxx.replace('$SDKTARGETSYSROOT', sdk_target_sysroot)
+                if re.match(r'export ARCH=', line):
+                    target_cpu = line[len(r'export ARCH='):]
+                    if target_cpu == 'arm64':
+                        arm_arch = 'armv8-a'
+                    elif target_cpu == 'arm':
+                        arm_arch = 'armv7ve'
+                if re.match(r'export CROSS_COMPILE=', line):
+                    cross_compile = line[len(r'export CROSS_COMPILE='):-1]
+                    print(cross_compile)
```

Comment:
Remove the debug print.

Reply:
also why -1 at the end but no +1 at start? This seems like off by one errors. Fix by using regex, shlex split

Reply:
the debug print is removed.
that -1 at the end is used to remove a hyphen. the CROSS_COMPILE value may be "arm-linux-", what is needed is "arm-linux".